### PR TITLE
Get annotations metadata csv

### PR DESCRIPTION
**Pull request by sofiacbeco**
Works for collections with different types of annotations.

### CODE REVIEW

**Comment by ahaith** [comment]
The diff here includes changes from #86 so it is difficult to review. Best to either wait until that PR is merged, or else rebase the relevant commits here onto master. We can walk you through this.

**Review by ahaith — APPROVED** [review]
Looks good 👍
FYI @Vaspra possible that some of your code using the SDK will need to add `anno_type` since the default is now to get all annotation types rather than just masks.

**Comment by sofiacbeco** [comment]
Great! I think there's one issue with get_annotations_for_image, as it is returning an empty list, and the 'row_index' input is not very clear to me.

File: zegami_sdk/collection.py
Comment (Vaspra) [inline zegami_sdk/collection.py:1037]: Unless you want to keep the DataFrame's index column, I usually add `index=False` to `DataFrame.to_csv` to avoid an unexpected column.

File: zegami_sdk/collection.py
Comment (Vaspra) [inline zegami_sdk/collection.py:1031]: Personally I'd avoid this method altogether from library as it doesn't bring functionality, if someone wants to save out the data they can use `c.get_annotations_metadata(anno_type).to_csv(path, index=False)`

Comment (ahaith) [reply zegami_sdk/collection.py:1031]: Yeah, I agree in fact. Limiting the surface area of the sdk does have benefits. Perhaps @sofiacbeco could remove in another PR?

File: zegami_sdk/collection.py
Comment (Vaspra) [inline zegami_sdk/collection.py:1028]: I'm not sure what this method is targeted at, uploaded classes are 0-indexed. Is this method referring to 1-indexed classes in a specific project? Typical class specification in a collection is:
```
[{'color': '#e6b3b3', 'id': '0', 'name': 'Damage'},
 {'color': '#bfb340', 'id': '1', 'name': 'Corrosion'},
 {'color': '#000000', 'id': '2', 'name': 'Dummy0'},
 {'color': '#000000', 'id': '3', 'name': 'Dummy1'},
 {'color': '#000000', 'id': '4', 'name': 'Dummy2'}]
```

File: zegami_sdk/collection.py
Comment (Vaspra) [inline zegami_sdk/collection.py:1041]: Has this been tested? On mine this does not function - for example `.get_annotations()` result has no `['annotations']` key
Is the metadata column re-ordering tuple specific to one collection?

Comment (Vaspra) [reply zegami_sdk/collection.py:1041]: Suggest replacing with
```
def get_annotations_as_dataframe(self, anno_type=None, source=0) -> pd.DataFrame:
    """
    Collects all annotations of a type (or all if anno_type=None) and returns
    the information as a dataframe.
    """
    
    annos = self.get_annotations(anno_type=anno_type)['sources'][source]['annotations']
    to_dict = lambda a: {
        'Row Index': self.imageset_index_to_row_index(a['image_index']),
        'ID': a['id'],
        'Type': a['type'],
        'Author': a['author'],
        'Imageset Index': a['image_index'],
    }
    annos_dictlist = [to_dict(a) for a in annos]
    
    df = pd.DataFrame(annos_dictlist)
    
    return df
```

Comment (sofiacbeco) [reply zegami_sdk/collection.py:1041]: Does this get the annotations['metadata'] as well? It doesn't seem to and it is important for Ebbon... This code was based and tested on annotations manually added on zegami

Comment (ahaith) [reply zegami_sdk/collection.py:1041]: The existing code looks ok, although Doug's suggestion is easier to follow. Would need to include the metadata as Sofia points out.

File: zegami_sdk/collection.py
Comment (ahaith) [inline zegami_sdk/collection.py:1054]: I agree that having the 'magic numbers' here doesn't seem quite right and won't work if the metadata isn't a specific length.
Better IMO to leave any ordering requirements to the 'client' script which receives this dataframe back.

**Pull request merged**
